src/drive/web/modules/views/Folder/syncHelpers.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Whether there is a file referenced by a share id
+ * @param {array} queryResults - List of folders and files
+ * @param {string} sharingId - Id of an io.cozy.sharings doc
+ * @returns {bool} true|false
+ */
+export const isThereFileReferencedBySharingId = (queryResults, sharingId) => {
+  return queryResults.some(query => {
+    return query.data.some(file => {
+      const fileReferences =
+        file.referenced_by && file.referenced_by.length >= 1
+      if (fileReferences) {
+        return file.referenced_by.some(reference => {
+          if (reference.type === 'io.cozy.sharings') {
+            return reference.id === sharingId
+          }
+          return false
+        })
+      }
+      return false
+    })
+  })
+}
+
+/**
+ * Remove a share from sharing context
+ * @param {object} params - Params
+ * @param {object} params.sharingsValue - Sharing Context value
+ * @param {function} params.setSharingsValue - Sharing Context setter
+ * @param {string} params.sharingId - Id of an io.cozy.sharings doc
+ */
+export const removeSharingFromContext = ({
+  sharingsValue,
+  setSharingsValue,
+  sharingId
+}) => {
+  delete sharingsValue[sharingId]
+  setSharingsValue(sharingsValue)
+}
+
+/**
+ * Create a syncing fake file
+ * @param {object} params - Params
+ * @param {string} params.sharingId - Id of an io.cozy.sharings doc
+ * @param {object} params.sharingsValue - Sharing Context value
+ * @param {object} params.fileValue - Sharing Context file value
+ * @returns {object} Syncing fake file
+ */
+export const createSyncingFakeFile = ({
+  sharingId,
+  sharingsValue,
+  fileValue
+}) => {
+  if (fileValue) {
+    return null
+  }
+
+  const sharingValue = sharingsValue[sharingId]
+
+  if (!sharingValue) {
+    return null
+  }
+
+  return {
+    name: sharingValue.attributes.description,
+    id: sharingValue.id,
+    type: 'directory'
+  }
+}
+
+/**
+ * Returns syncingFakeFile if it is still needed, otherwise remove the share in context
+ * @param {object} params - Params
+ * @param {array} params.queryResults - List of folders and files
+ * @param {string} params.sharingId - Id of an io.cozy.sharings doc
+ * @param {object} params.sharingsValue - Sharing Context value
+ * @param {function} params.setSharingsValue - Sharing Context setter
+ * @param {object} params.syncingFakeFile - Syncing fake file
+ * @returns {object} Syncing fake file
+ */
+export const checkSyncingFakeFileObsolescence = ({
+  queryResults,
+  sharingId,
+  sharingsValue,
+  setSharingsValue,
+  syncingFakeFile
+}) => {
+  const isThereRealtimeFileReferencedBySharing = isThereFileReferencedBySharingId(
+    queryResults,
+    sharingId
+  )
```

Comment:
Since this check can be "heavy" (we can use it on thousand of documents) we should memoize the call in order to try to prevent to call it too ofter

Reply:
Could reselect be used ?

Reply:
After reading again the code, this method is only called when a sharing is in progress. So not so much, so it's not as much as important as I thought at the beginning. 

The question / the suggestion remains however interesting:
- to me reselect helps to read data from redux store. Today the data we read are coming from useQuery hooks and not directly from the store itself. Even if it can be possible to read directly from the store, I don't know if this is useful in that case. Maybe to reduce the number of args for a few methods, but it will add a direct dep to the store. I don't have strong opinion on that. 

- I don't think that in this particular case, reselect would be more useful than useMemo. To me (no time passed on it but...) reselect should be connected on FolderView to have access to the queryResult and "block" re-renders if needed. But we have also state on this component so the render will be done even if the store doesn't change and so the call to useSyncingFakeFile(). So I think it's better to memoize the method call. 

Don't know if it's clear enough 🤔

Reply:
Oui c'est exécuté après validation d'un partage, et seulement tant qu'il y a un partage en cours. Je memoise le resultat 👍